src/wh_server_keystore.c

```diff
@@ -239,11 +243,18 @@ int hsmReadKey(whServerContext* server, whKeyId keyId, whNvmMetadata* outMeta,
         hsmCacheKey(server, meta, out);
     }
 #ifdef WOLFHSM_SHE_EXTENSION
-    /* use empty string if we couldn't find the master ecu key */
-    if (ret != 0 && keyId == WOLFHSM_SHE_MASTER_ECU_KEY_ID) {
+    /* use empty key of zeros if we couldn't find the master ecu key */
+    if (ret == WH_ERROR_NOTFOUND &&
+        (keyId & WOLFHSM_KEYTYPE_MASK) == WOLFHSM_KEYTYPE_SHE &&
+        (keyId & WOLFHSM_KEYID_MASK) == WOLFHSM_SHE_MASTER_ECU_KEY_ID) {
         XMEMSET(out, 0, WOLFHSM_SHE_KEY_SZ);
-        meta->len = WOLFHSM_SHE_KEY_SZ;
-        meta->id = searchId;
+        *outSz = WOLFHSM_SHE_KEY_SZ;
+        if (outMeta != NULL) {
+            /* need empty flags and corect lenth and id */
+            XMEMSET(outMeta, 0, sizeof(meta));
+            meta->len = WOLFHSM_SHE_KEY_SZ;
+            meta->id = keyId;
+        }
         ret = 0;
     }
 #endif
```

Comment:
Shouldn't this be handled in the SHE version of read key and not in the generic one?

Reply:
multiple places can read the unset ecu key, they don't call their own she read function they make a she keyId and then call the normal read function, putting it here means no duplicated code